### PR TITLE
Implement read-only option to playground sessions

### PR DESCRIPTION
**Pull request by izruff**
### Description

Adds an option to invite other users to a session as viewers (read-only mode) or editors.

### Changes to implementation
- Users can join a session using one of two IDs: the session editing ID or the session viewing ID. These IDs are only 6 characters long, allowing for easier sharing.
- URL paths for requests sent to the ShareDB backend now contain only the 6-character ID instead of the ShareDB document ID used in previous implementations (see [this PR](https://github.com/source-academy/sharedb-ace-backend/pull/25)).
- Details of the current session such as the document ID and read-only status are obtained by GET requests, stored in Redux, and then passed to the ShareDB-Ace client (see [this PR](https://github.com/source-academy/sharedb-ace/pull/1)).

### UI changes
- Previously, clicking on the invite button immediately creates a session. Now it shows either a Create button, or the session IDs if they are currently in a session.
- Upon successfully joining a session, user is greeted with a popup message stating their level of access (as an editor or viewer).
- Added labels to text inputs.

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Code quality improvements

### How to test

To create a new session, click on Invite > Create. Open SA Playground in another tab and join using one of the session IDs (Join > enter your ID), after which a popup message will appear. If joining as editor, the Ace editor still works as normal. If joining as a viewer, however, the editor will be set to read-only, meaning users cannot change the text as if it were a text field. But users can still move their cursor or block parts of the code in the editor.

### Checklist

- [x] I have tested this code
- [ ] I have updated the documentation

### CODE REVIEW

File: src/commons/collabEditing/CollabEditingHelper.ts [hunk 1]
```diff
@@ -13,17 +13,33 @@ export function getSessionUrl(sessionId: string, ws?: boolean): string {
   return url.toString();
 }
 
+/*
 export async function checkSessionIdExists(sessionId: string): Promise<boolean> {
   const resp = await fetch(getSessionUrl(sessionId));
 
   return resp && resp.ok;
 }
+*/
```
Comment (RichDom2185) [inline src/commons/collabEditing/CollabEditingHelper.ts:22]: It seems like we no longer need this function. Can we delete it?

**Review by RichDom2185 — COMMENTED** [review]
Hi, overall looks good. Just a minor comment below, as we as another question:

Do we need to bump the `@sourceacademy/sharedb-ace` package in `package.json` for this frontend PR to work?

File: src/commons/collabEditing/CollabEditingHelper.ts (hunk 1 above)
Comment (izruff) [reply src/commons/collabEditing/CollabEditingHelper.ts]: Yes.

**Comment by izruff** [comment]
> Do we need to bump the `@sourceacademy/sharedb-ace` package in `package.json` for this frontend PR to work?

Yes, it is necessary for the read-only functionality (disabling the editor in viewer mode) to work.

**Comment by RichDom2185** [comment]
> > Do we need to bump the `@sourceacademy/sharedb-ace` package in `package.json` for this frontend PR to work?
> 
> Yes, it is necessary for the read-only functionality (disabling the editor in viewer mode) to work.

Ok, will mark as blocked in the meantime. Once the `@sourceacademy/sharedb-ace` is updated and a new version is pushed to NPM, please make the necessary updates to `package.json` and `yarn.lock`.

Thanks!

**Comment by RichDom2185** [comment]
> > Do we need to bump the `@sourceacademy/sharedb-ace` package in `package.json` for this frontend PR to work?
> 
> Yes, it is necessary for the read-only functionality (disabling the editor in viewer mode) to work.

Update: As discussed via Telegram, a `@sourceacademy/sharedb-ace` update is no longer necessary.

**Review by RichDom2185 — APPROVED** [review]
LGTM, thanks! As per discussion on Telegram, we will live with the breaking change until we can redeploy the backend PR.

**Pull request merged**